### ISSUE TITLE
Referencing package Microsoft.Orleans.OrleansTelemetryConsumers.Counters with Orleans 7

### ISSUE THREAD

**Issue by gparissis** [dotnet/orleans#8117]
The latest version of the package is 3.6.5.
When I reference it in my project I get the below error:

> Error	CS7069	Reference to type 'HostBuilderContext' claims it is defined in 'Orleans.Core', but it could not be found

Is this package going to be updated or it has been replaced by another one?

Thank you.

**Comment by ReubenBond** [comment]
It is obsolete now, you can use OpenTelemetry or vendor specific libraries. I showed a quick demo in this .NET Conf Session: https://youtu.be/WUZw78P-Zg8

The demo is right at the end. I'm hoping we can have a sample + docs for that soon

**Comment by tomasevicius** [comment]
So...
I had a simple line of code like this:
siloBuilder.AddApplicationInsightsTelemetryConsumer(myAzureInsightsKey);

How should I change this code to have logging in AI with Orleans v7?

**Comment by ReubenBond** [comment]
@tomasevicius if you want logs to appear in Application Insights (i.e, textual logs), then I imagine this package is the one your should use: https://www.nuget.org/packages/Microsoft.Extensions.Logging.ApplicationInsights
This documentation page shows how to configure the host for Application Insights logging: https://learn.microsoft.com/en-us/azure/azure-monitor/app/ilogger#example-programcs

**Comment by tomasevicius** [comment]
Hi, I also get the same error "Reference to type 'HostBuilderContext' claims it is defined in 'Orleans.Core', but it could not be found" in ServiceFabric service project.
That is obviously because of "<PackageReference Include="Microsoft.Orleans.Hosting.ServiceFabric" Version="3.6.5" />".
Removing this reference and keeping only "Microsoft.Orleans.Server" does not help as "OrleansServiceListener" static class is not found.

The documentation related to hosting in ServiceFabric with Orleans v7 is not updated yet.. :(
https://learn.microsoft.com/en-us/dotnet/orleans/deployment/service-fabric

**Comment by tomasevicius** [comment]
@ReubenBond Hi, I see that documentation was updated on how to host with ServiceFabric.
Unfortunately, this line of sample code throws an exception:
`await ServiceRuntime.RegisterServiceAsync("Orleans.ServiceFabric.Stateless", context => new OrleansHostedStatelessService(CreateHostAsync, context));`

Exception:
`System.Fabric.FabricException
  HResult=0x80071C21
  Message=Invalid Service Type
  ErrorCode=System.Fabric.FabricErrorCode.InvalidServiceType
  Source=Microsoft.ServiceFabric.Services`
  
 I am using the latest "Microsoft.ServiceFabric.Services" v6.1.1436, "Microsoft.Orleans.Server" v7.0.0.
 I have locally installed MS Azure Service Fabric SDK v6.1.1436.9590 + Runtime v9.1.1436.9590.
 
 I assume value "Orleans.ServiceFabric.Stateless" as parameter for "serviceTypeName" is wrong.

**Comment by ReubenBond** [comment]
@tomasevicius where did you obtain the sample from? Did you recreate it from scratch or use the version included in the docs page (here: https://github.com/dotnet/docs/tree/main/docs/orleans/deployment/snippets/service-fabric)?

**Comment by tomasevicius** [comment]
@ReubenBond I am migrating from Orleans v3.6 to v7.0, so I already had Service project. I was reading the updated documentation here: https://learn.microsoft.com/en-us/dotnet/orleans/deployment/service-fabric.
After looking at the github sample you gave, I figured out that I need to change "Orleans.ServiceFabric.Stateless" to the value used in my ServiceManifext.xml.

Value "Orleans.ServiceFabric.Stateless" seemed like a special/const value to use, I did not realise I need to change that :)
But it works now, thank You for help!
